x-pack/plugins/secret_service/server/service/secret_service.test.ts

```diff
@@ -0,0 +1,179 @@
+  const savedObjectsClient = {
+    create: jest.fn(),
+    errors: {
+      isConflictError: jest.fn(),
+    },
+    bulkCreate: jest.fn(),
+    bulkGet: jest.fn(),
+    delete: jest.fn(),
+    find: jest.fn(),
+    get: jest.fn(),
+    update: jest.fn(),
+  };
+
+  const subject = new SecretService(savedObjectsClient as any, 'testSecretType');
+
+  it('should not expose the password field', () => {
+    const field = 'password';
+    const sameSubject: any = subject;
+    expect(sameSubject[field]).toBeUndefined();
+  });
+
+  it('should hide values using the saved object client', async () => {
+    const hideMe = { message: 'my secret message', anotherSecret: 'this is unhidden' };
+    let insideJob: any = null;
+    savedObjectsClient.create.mockImplementation((type: string, attributes: any, options?: any) => {
+      insideJob = {
+        id: options.id,
+        type: 'testSecretType',
+        attributes,
+      };
+      return insideJob;
+    });
+
+    const hidden = await subject.hideAttribute(hideMe);
+    expect(savedObjectsClient.create).toHaveBeenCalledTimes(1);
+    expect(savedObjectsClient.update).not.toHaveBeenCalled();
+    expect(hidden).toBeDefined();
+    expect(typeof hidden).toBe('string');
+    expect(hidden).toMatch(/^[\w+\/]+/);
+    expect(subject);
+  });
+
+  it('should be able to unhide values from saved objects', async () => {
+    expect.assertions(3);
+    const hideMe = {
+      message: 'my secret message',
+      anotherSecret: 'this is unhidden',
+      someOtherProp: 'check me',
+    };
+    let insideJob: any = null;
+    savedObjectsClient.create.mockImplementation((type: string, attributes: any, options?: any) => {
+      insideJob = {
+        id: options.id,
+        type: 'testSecretType',
+        attributes,
+      };
+      return insideJob;
+    });
+    savedObjectsClient.get.mockImplementation((type: string, id: string, options?: any) => {
+      return insideJob;
+    });
+    const hidden = await subject.hideAttribute(hideMe);
+    const unhidden = await subject.unhideAttribute(hidden);
+    expect(unhidden.id).toEqual(hidden);
+    expect(unhidden.type).toEqual('testSecretType');
+    expect(unhidden.attributes).toMatchObject(hideMe);
+  });
+
+  it('should be unable to unhide values if the saved object was changed in any way', async () => {
+    expect.assertions(1);
+    const hideMe = {
+      message: 'my secret message',
+      anotherSecret: 'this is unhidden',
+      someOtherProp: 'check me',
+    };
+    let insideJob: any = null;
+    const internalSavedObject: any = null;
+    savedObjectsClient.create.mockImplementation((type: string, attributes: any, options?: any) => {
+      insideJob = {
+        id: 'testId',
+        type: 'testSecretType',
+        attributes,
+      };
+      return insideJob;
+    });
+    savedObjectsClient.get.mockImplementation((type: string, id: string, options?: any) => {
```

Comment:
__nit:__ maybe something like this? I don't see we reuse `internalSavedObject` anywhere.
```ts
savedObjectsClient.get.mockResolvedValue({
  anotherSecret: 'I am changing the message',
  addNewField: 'me',
});
```